src/extension/serializer.ts

```diff
@@ -353,6 +353,11 @@ export abstract class SerializerBase implements NotebookSerializer, Disposable {
     }
 
     data.cells.forEach((cell) => {
+      if (identity === RunmeIdentity.UNSPECIFIED && cell.metadata?.['runme.dev/stageId']) {
```

Comment:
@degrammer During the deserialization/serialization/save process, the **stageId** is used to transport the ID and then recover it to its original form as the ID. Instead of "stage" it could be called a "snapshot," "copy," or "tmp," which sounds clearer to me. However, I will explore finding a new, more descriptive field name.

Reply:
@degrammer, I updated it to "originalId".